File: examples/benchmark/writer.rb
require 'rubygems'
require 'karait'

puts 'Starting ruby writer.'

messages_written = 0
start = Time.new.to_f
queue = Karait::Queue.new

while true
    queue.write({
        'messages_written' => messages_written,
        'sender' => 'writer.rb',
        'started_running' => start,
        'messages_written_per_second' => (messages_written / (Time.new.to_f - start))
    }, :routing_key => 'for_reader')
    messages_written += 1
end

File: examples/benchmark/writer.js
var puts = require('sys').puts,
    Queue = require('karait').Queue;

puts("Starting javascript writer.")

messagesWritten = 0.0
startTime = (new Date()).getTime() / 1000.0;

new Queue(function(err, queue) {
    if (err) {
        throw err;
    }
    
    (function writeMessage() {
        queue.write(
            {
                messages_written: messagesWritten,
                sender: 'writer.js',
                started_running: startTime,
                messages_written_per_second: messagesWritten / ( ( (new Date()).getTime() / 1000.0 ) - startTime )
            },
            {
                routingKey: 'for_reader'
            },
            function() {
                writeMessage();
            }
        )
        messagesWritten += 1;
    })();
});